### ISSUE TITLE
Breaking error around "open /proc/sys/net/ipv4/conf/eth0/rp_filter: no such file or directory" in 1.8.3

### ISSUE THREAD

**Issue by mvisonneau** [cilium/cilium#13143]
Originally discussed in https://github.com/cilium/cilium/issues/10645#issuecomment-688446819, @joestringer advised to open a separate issue for this behaviour/bug which seems to be a regression.

## Bug report

After attempting a `1.8.2` -> `1.8.3` upgrade on **Ubuntu 20.04 / 5.4.0-1021-aws**, agent pods end up crashing with the following logs. Rolling back to 1.8.2, pods are healthy and do not contain those errors. 🤷

```
{"error":"Failed to sysctl -w net.ipv4.conf.eth0.rp_filter=2: could not open the sysctl file /proc/sys/net/ipv4/conf/eth0/rp_filter: open /proc/sys/net/ipv4/conf/eth0/rp_filter: no such file or directory","level":"error","msg":"Error while initializing daemon","subsys":"daemon"}
{"error":"Failed to sysctl -w net.ipv4.conf.eth0.rp_filter=2: could not open the sysctl file /proc/sys/net/ipv4/conf/eth0/rp_filter: open /proc/sys/net/ipv4/conf/eth0/rp_filter: no such file or directory","level":"fatal","msg":"Error while creating daemon","subsys":"daemon"}
{"error":"Operation cannot be fulfilled on ciliumnodes.cilium.io \"ip-10-6-11-13.eu-west-1.compute.internal\": the object has been modified; please apply your changes to the latest version and try again","level":"warning","msg":"Unable to update CiliumNode custom resource","subsys":"ipam"}
{"level":"info","msg":"regenerating all endpoints","reason":"one or more identities created or deleted","subsys":"endpoint-manager"}
{"level":"info","msg":"regenerating all endpoints","reason":"one or more identities created or deleted","subsys":"endpoint-manager"}
```
**General Information**

- Cilium version: `1.8.3`
- Kernel version: `5.4.0-1021-aws`
- Orchestration system version: `EKS : v1.17.9-eks-4c6976`

**How to reproduce the issue**

1. [TO BE CONFIRMED] -> run an EKS cluster with [AWS Nitro](https://aws.amazon.com/ec2/nitro/) instances, [presumably](https://github.com/cilium/cilium/issues/10645#issuecomment-689025791) the issue is around the lack of `eth0` interface 
2. It should work with 1.8.2 but upgrading to 1.8.3 would cause the Cilium agent to crash after a few seconds

**Comment by joestringer** [comment]
Looks related to https://github.com/cilium/cilium/pull/12770, which I think was tested in standard EKS with Amazon Linux 2, which provides the eth0 interface by default.

**Comment by joestringer** [comment]
Looks like it affects v1.7.8, v1.7.9 as well.

**Comment by joestringer** [comment]
Also relevant, interface listing from @mvisonneau on the other issue:
```
~$ netstat -i
Kernel Interface table
Iface      MTU    RX-OK RX-ERR RX-DRP RX-OVR    TX-OK TX-ERR TX-DRP TX-OVR Flg
cilium_h  9001        0      0      0 0             0      0      0      0 BMORU
cilium_n  9001        0      0      0 0             0      0      0      0 BMORU
docker0   1500        0      0      0 0             0      0      0      0 BMU
ens5      9001 77873091      0      0 0      114673441      0      0      0 BMRU
ens6      9001 605028990      0     74 0      495146454      0      0      0 BMRU
ens7      9001 64204300      0      0 0      59610683      0      0      0 BMRU
lo       65536 135179116      0      0 0      135179116      0      0      0 LRU
lxc84712  9001  5109046      0      0 0       5763573      0      0      0 BMRU
lxc053a0  9001  1566665      0      0 0       1104132      0      0      0 BMRU
lxc17cb3  9001   280988      0      0 0        251064      0      0      0 BMRU
lxc225b5  9001  6816453      0      0 0       5947230      0      0      0 BMRU
lxc25c93  9001  3330495      0      0 0       3939436      0      0      0 BMRU
lxc2ae26  9001   116270      0      0 0        141742      0      0      0 BMRU
lxc3b24a  9001 205922198      0      0 0      263466611      0      0      0 BMRU
lxc3d661  9001  4308719      0      0 0       6791656      0      0      0 BMRU
lxc45541  9001  2294528      0      0 0       2473290      0      0      0 BMRU
lxc49be6  9001  3975092      0      0 0       2414246      0      0      0 BMRU
lxc5f6ec  9001   148957      0      0 0        148039      0      0      0 BMRU
lxc676a3  9001  1057937      0      0 0        645921      0      0      0 BMRU
lxc67811  9001 236109688      0      0 0      206980719      0      0      0 BMRU
lxc6b001  9001   169636      0      0 0        168709      0      0      0 BMRU
lxc75d86  9001 19918743      0      0 0      16686086      0      0      0 BMRU
lxc7d310  9001 70644814      0      0 0      47580110      0      0      0 BMRU
lxc_heal  9001  1628264      0      0 0       1932101      0      0      0 BMRU
lxcb9ade  9001  2623227      0      0 0       3048972      0      0      0 BMRU
lxcc56bb  9001 15626813      0      0 0      21041815      0      0      0 BMRU
lxccd212  9001   234663      0      0 0        305566      0      0      0 BMRU
lxcd656e  9001   145880      0      0 0        145195      0      0      0 BMRU
lxcd7296  9001  5129261      0      0 0       4645878      0      0      0 BMRU
lxcdecd9  9001  2764864      0      0 0       3648134      0      0      0 BMRU
lxce1a74  9001    32706      0      0 0         32298      0      0      0 BMRU
lxcef306  9001 21973644      0      0 0      21831440      0      0      0 BMRU
lxcf535c  9001 24108700      0      0 0      27435746      0      0      0 BMRU
```

**Comment by ungureanuvladvictor** [comment]
I hit this as well with the upstream kernel and had to add the following to the GRUB cmdline `net.ifnames=0 biosdevname=0` to fallback to old device names.

**Comment by joestringer** [comment]
From Slack discussion, it looks like the logic for this piece is relying on `eth0` directly, but we could likely make it a bit more intelligent to detect the device with the default route to apply the logic to that device instead (via `NodeDeviceWithDefaultRoute`). This should allow the logic to work correctly on any Linux distribution without reconfiguring the way that your kernel assigns interface names.

**Comment by chowmean** [comment]
@joestringer I am also facing this issue what can be done to fix this? 
OS: Ubuntu 18.04.4
SystemD version: 237
Kubernetes: EKS 1.17

**Comment by joestringer** [comment]
@chowmean workarounds right now include either configuring the kernel to ensure `eth0` is present per @ungureanuvladvictor 's comment above, or sticking with v1.8.2 for now. This is planned to be fixed in the upcoming v1.8.4 release.

**Comment by qmonnet** [comment]
Heads up: I've had a patchset ready for a few days, but I failed to set up a cluster to test it properly because of #13193. This should be fixed in the next few days and I'll test and send my PR.
